poetry/console/config/application_config.py

```diff
@@ -218,7 +220,19 @@ def create_io(
             Output(error_stream, error_formatter),
         )
 
-        resolved_command = application.resolve_command(args)
+        try:
+            resolved_command = application.resolve_command(args)
+        except NoSuchOptionException as ex:
```

Comment:
Should we expand to this to all similar exceptions?

Reply:
Do you mean that I should catch bare `Exception` there? I didn't find any root `clickit` exception, which I can catch there